xrpl4j-core/src/main/java/org/xrpl/xrpl4j/crypto/keys/Seed.java

```diff
@@ -386,11 +389,24 @@ private static KeyPair deriveKeyPair(final UnsignedByteArray seedBytes, final in
         final BigInteger privateKeyInt = derivePrivateKey(seedBytes, accountNumber);
         final UnsignedByteArray publicKeyInt = derivePublicKey(privateKeyInt);
 
+        // Both ed25519 and secp256k1 _private_ keys are always 32 bytes long. However, in this library, both types of
+        // private key are prefixed with one byte (0xED for ed25519 and 0x00 for secp256k1) because this is what other
+        // libraries do, and also so that any particular set of 32 private key bytes can be properly disambiguated.
+        // See https://github.com/XRPLF/xrpl4j/issues/486 for more details.
+        final UnsignedByteArray prefixedPrivateKey;
+        if (privateKeyInt.toByteArray().length == 32) {
+          prefixedPrivateKey = UnsignedByteArray.of(PrivateKey.SECP256K1_PREFIX)
+            .append(UnsignedByteArray.of(privateKeyInt.toByteArray()));
+        } else {
+          prefixedPrivateKey = UnsignedByteArray.of(privateKeyInt.toByteArray());
+        }
+
+        // No need to prefix secp256k1 public keys because they are always 33 bytes.
+        final UnsignedByteArray prefixedPublicKey = UnsignedByteArray.of(publicKeyInt.toByteArray());
+
         return KeyPair.builder()
-          .privateKey(PrivateKey.of(UnsignedByteArray.of(privateKeyInt.toByteArray())))
-          .publicKey(PublicKey.fromBase16EncodedPublicKey(
-            UnsignedByteArray.of(publicKeyInt.toByteArray()).hexValue()
-          ))
+          .privateKey(PrivateKey.of(prefixedPrivateKey))
+          .publicKey(PublicKey.fromBase16EncodedPublicKey(prefixedPublicKey.hexValue()))
```

Comment:
Should we call `PublicKey.builder().value(prefixedPublicKey).build()` here rather than encode to hex and then parse from hex?

Reply:
Turns out we can just use `publicKeyInt` directly. Fixed in 68b3292cfab0eb332263f7c40b9a51c44d344b1b